client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcConnection.java

```
@@ -670,4 +684,20 @@ public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, Connect
             return table.build().cursor();
         }
     }
+
+    private static void assertThatFutureIsBlocked(Future<?> future)
+    {
+        if (!future.isDone()) {
+            return;
+        }
+        try {
+            future.get();
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+        finally {
+            fail("expecting all futures to be blocked");
```

that swallows the exception raised 3 lines above